test/utils/shoots/access/adminkubeconfig.go

```diff
@@ -17,31 +17,26 @@ package access
 import (
 	"context"
 
-	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/utils/pointer"
 
 	authenticationv1alpha1 "github.com/gardener/gardener/pkg/apis/authentication/v1alpha1"
 	gardencorev1beta1 "github.com/gardener/gardener/pkg/apis/core/v1beta1"
-	gardenversionedcoreclientset "github.com/gardener/gardener/pkg/client/core/clientset/versioned"
 	"github.com/gardener/gardener/pkg/client/kubernetes"
 )
 
 // CreateShootClientFromAdminKubeconfig requests an admin kubeconfig and creates a shoot client.
 func CreateShootClientFromAdminKubeconfig(ctx context.Context, gardenClient kubernetes.Interface, shoot *gardencorev1beta1.Shoot) (kubernetes.Interface, error) {
-	versionedClient, err := gardenversionedcoreclientset.NewForConfig(gardenClient.RESTConfig())
-	if err != nil {
-		return nil, err
-	}
-
 	adminKubeconfigRequest := &authenticationv1alpha1.AdminKubeconfigRequest{
 		Spec: authenticationv1alpha1.AdminKubeconfigRequestSpec{
 			ExpirationSeconds: pointer.Int64(3600),
 		},
 	}
-	adminKubeconfig, err := versionedClient.CoreV1beta1().Shoots(shoot.GetNamespace()).CreateAdminKubeconfigRequest(ctx, shoot.GetName(), adminKubeconfigRequest, metav1.CreateOptions{})
-	if err != nil {
+	if err := gardenClient.Client().SubResource("adminkubeconfig").Create(ctx, shoot, adminKubeconfigRequest); err != nil {
+		return nil, err
+	}
+	if err := gardenClient.Client().SubResource("adminkubeconfig").Get(ctx, shoot, adminKubeconfigRequest); err != nil {
```

Comment:
What's this, why do we need another `Get` call here? `Create` should already return the kubeconfig in the `adminKubeconfigRequest.Status`?